server/sonar-web/src/main/js/apps/users/components/TokensForm.tsx

```diff
@@ -108,9 +103,17 @@ export default class TokensForm extends React.PureComponent<Props, State> {
     }
   };
 
-  handleRevokeToken = () => {
-    this.setState({ hasChanged: true });
-    this.fetchTokens();
+  handleRevokeToken = (revokedToken: UserToken) => {
+    this.setState(
+      state => ({
+        tokens: state.tokens.filter(token => token.name !== revokedToken.name)
+      }),
+      () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update this function so that its implementation is not identical to the one on line 89. ([typescript:S4144](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS4144))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Ftokens&issues=AWDgxWXTfL4FhfRSWp_Q&open=AWDgxWXTfL4FhfRSWp_Q)

[ISSUE_KEY]: AWDgxWXTfL4FhfRSWp_Q